### PR TITLE
feat: track no-shows in bookings

### LINKED ISSUE #13677: [CAL-3134] ability to mark guests as no-show
image

click will open a dropdown:

* "copy email"
* "mark as no-show"

image

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3134](https://linear.app/calcom/issue/CAL-3134/ability-to-mark-guests-as-no-show)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
## What does this PR do?

- For 1 & 2 guests only (where you see name/email on the booking list item) you can click on the name/email to see a dropdown

> [!NOTE]
> All no-show related actions are only visible/doable in the bookings/past page, as one might expect

- Added functional Email & Copy Email To Clipboard action on each booking page (1-2 attendees)
    image

- Adds functional Guest No-Show feature on the "past" bookings page
    image

- For 1 & 2 guests only (where you see name/email on the booking list item) you can click on the name/email to see "mark as no-show" or "unmark as no-show" option depending upon the existing state of their no-show.
  image

- When marking/unmarking an individual user as no-show, you will see their email as being marked as no-show in the toast
  image

- When there are more than 2 attendees, we have `& <n> more` showing up in the booking item, you can now click on them and it loads up a dropdown containing a list of emails of all remaining attendees/guests. You can select which ones among those were a no-show, then proceed to click on "Mark as no show" which then updates their no-show state. TO emphasize **Checked = No-Show** we also put a strikethrough the email.
  image

- Since this action can be to both mark & unmark multiple attendees, and simultaneously act to both mark some and unmark others, the notifications states "No-Show status updated for attendees". We can improve this if there's a better alternative to this.
  image

[Loom](https://www.loom.com/share/141143be08874d3c8bd6747c17328d1d)

> [!NOTE]  
> Currently we don't wait to receive confirmation on the state of update, we just send the update call and are optimistic it's done. This can be changed if necessary.

## What remains / Next steps

- [ ] Tests
- [ ] Adding to insights

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/86) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

Fixes #13677 
Fixes CAL-3134

### CODE REVIEW

File: apps/web/components/booking/BookingListItem.tsx [hunk 1]
```diff
@@ -656,10 +674,60 @@ type AttendeeProps = {
...
+        </button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent>
+        {/* TODO: figure out how to close the dropdown after clicking, since we are stopping propagation due to the <Link> */}
+        <DropdownMenuItem className="focus:outline-none">
+          {/* TODO: add subject: title */}
+          <DropdownItem StartIcon={Mail} href={`mailto:${email}`} onClick={(e) => e.stopPropagation()}>
+            <a href={`mailto:${email}`}>{t("email")}</a>
+          </DropdownItem>
+        </DropdownMenuItem>
```
Comment (sean-brydon) [inline apps/web/components/booking/BookingListItem.tsx:711]: Interesting - i think we can use DropdownItem and pass href in without needing to put a link inside. Let me know if this doesnt work.

```suggestion
        <DropdownMenuItem className="focus:outline-none">
          {/* TODO: add subject: title */}
          <DropdownItem StartIcon={Mail} href={`mailto:${email}`} >
           {t("email")}
          </DropdownItem>
        </DropdownMenuItem>
```
If this doesnt work we may need to implement the "asChild" prop for DropdownItem

Comment (PeerRich) [reply apps/web/components/booking/BookingListItem.tsx:711]: thats fine

**Comment by shirazdole** [comment]
Can we add a webhook here?

**Comment by alishaz-polymath** [comment]
The noShow option should be in the past bookings, not in the upcoming no? @shirazdole @PeerRich

File: packages/ui/components/form/dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -71,12 +71,34 @@ export const DropdownMenuGroup = DropdownMenuPrimitive.Group;
...
+        )}
+        <DropdownMenuPrimitive.ItemIndicator asChild>
+          <input
+            aria-disabled={true}
+            aria-readonly
+            aria-label={typeof children === "string" ? `Active${children}` : undefined}
+            checked={true}
+            type="checkbox"
+            className="text-emphasis dark:text-muted focus:ring-emphasis border-default bg-default h-4 w-4 rounded hover:cursor-pointer"
+          />
```
Comment (alishaz-polymath) [inline packages/ui/components/form/dropdown/Dropdown.tsx:101]: Finally checkbox works as expected with the radix dropdown. Thank you @sean-brydon for fixing the design of this 🙏

**Pull request closed without merging**